core/src/main/java/com/sequenceiq/cloudbreak/reactor/BootstrapNewNodesHandler.java

```diff
@@ -28,13 +36,18 @@ public String selector() {
 
     @Override
     public void accept(Event<BootstrapNewNodesRequest> event) {
+        int numHosts = event.getData().getHostNames() == null ? -1 : event.getData().getUpscaleCandidateAddresses().size();
```

Comment:
you can use `Optional.fromNullable` here

Reply:
also `event.getData().getHostNames()` is checked if null, but `event.getData().getUpscaleCandidateAddresses()` size is used, so something is off here